rscripts/YAL5-single-codons.R

```diff
@@ -0,0 +1,1108 @@
+#TEST: ExpandedRegionNormalization(): the column "Rel_Pos" goes from
+#      -"expand_width to +"expand_width" 
+#TEST: ExpandedRegionNormalization(): sum(normalized_expand_list[[1]]$RelCount)
+#      /nrow(normalized_expand_list[[1]]) == 1 
+#TEST: ExpandedRegionNormalizetion(): None of the RelCount columns should contain NaN. 
+#      These should all be set to 0
+#
+# Example: using tinysim
+# A tibble: 3 x 6
+# Gene  PosCodon Count Codon Rel_Pos RelCount
+# <chr>    <dbl> <dbl> <chr>   <int>    <dbl>
+#   1 MAT          1     0 ATG        -1      0  
+# 2 MAT          2     2 GCC         0      1.5
+# 3 MAT          3     2 ACA         1      1.5
+#
+# After editing tinysim codon position 4 to inculde a GCC, 
+# the following output_feature_info object is produced
+# 
+# > output_feature_info
+# [[1]]
+# # A tibble: 3 x 5
+# Gene  PosCodon Count Codon Rel_Pos
+# <chr>    <dbl> <dbl> <chr>   <int>
+#   1 MAT          1     0 ATG        -1
+# 2 MAT          2     2 GCC         0
+# 3 MAT          3     2 ACA         1
+# 
+# [[2]]
+# # A tibble: 3 x 5
+# Gene  PosCodon Count Codon Rel_Pos
+# <chr>    <dbl> <dbl> <chr>   <int>
+#   1 MIKE         3     0 AAG        -1
+# 2 MIKE         4     0 GCC         0
+# 3 MIKE         5     0 TAA         1
+#
+# This is processed by ExpandedRegionNormalization to produce the following output
+
+# > normalized_expand_list
+# [[1]]
+# # A tibble: 3 x 6
+# Gene  PosCodon Count Codon Rel_Pos RelCount
+# <chr>    <dbl> <dbl> <chr>   <int>    <dbl>
+#   1 MAT          1     0 ATG        -1      0  
+# 2 MAT          2     2 GCC         0      1.5
+# 3 MAT          3     2 ACA         1      1.5
+# 
+# [[2]]
+# # A tibble: 3 x 6
+# Gene  PosCodon Count Codon Rel_Pos RelCount
+# <chr>    <dbl> <dbl> <chr>   <int>    <dbl>
+#   1 MIKE         3     0 AAG        -1        0
+# 2 MIKE         4     0 GCC         0        0
+# 3 MIKE         5     0 TAA         1        0
+# 
+
+
+# Normalization carried out for all the tibbles within ExpandList 
+# normalized_expand_list <- purrr::map(
+#   .x = output_feature_info,
+#   .f = ExpandedRegionNormalization,
+#   expand_width
+# )
+
+
+# TEST:: normalised_expand_list should be of type list
+# type(normalized_expand_list)
+# [1] "list"
+
+# TEST:: Normalized_expand_list should be the same length as expand_feature_region
+# > length(normalized_expand_list)==length(expand_feature_region)
+# [1] TRUE
+
+# TEST:: the dimensions of each item in the list shoud be [(2*expand_width+1) X 5]
+# as there are now 5 rows; Gene, Pos_Codon, Rel_Count, Rel_Pos, RelCount
+# > dim(expand_feature_region[[1]])
+# [1] 11  5
+
+# TEST:: At each position, the sum of RelCount should be equal to (2*expand_width+1)
+# ie if the expand width was 5:
+# sum(normalized_expand_list[[1]]$RelCount)
+# [1] 11
+
+
+
+
+### Functions for overlaying the normalized expanded tibbles ###
+
+# Function to overlay graphs into a single graph. Need to generate a single tibble 
```

Comment:
Inaccurate, this function overlays the tibbles so that a graph can be made later in the script